src/main/java/net/minecraftforge/event/entity/SpawnPlacementRegisterEvent.java

```diff
@@ -0,0 +1,204 @@
+ *
+ * If modifying vanilla or another mod's spawn placements, you can use three operations:
+ *  REPLACE: checked first, the last mod to replace the predicate wipes out all other predicates. Listen with a low {@link EventPriority} if you need to do this.
+ *  OR: checked second, only one of these predicates must pass along with the original predicate
+ *  AND: checked third, these predicates must all pass along with the original predicate
+ *
+ * <p>
+ * This event is not {@linkplain Cancelable cancellable} and does not {@linkplain Event.HasResult have a result}.
+ * <p>
+ *
+ *  Fired on the Mod bus {@link IModBusEvent}.<br>
+ */
+public class SpawnPlacementRegisterEvent extends Event implements IModBusEvent
+{
+    private final Map<EntityType<?>, MergedSpawnPredicate> map;
+
+    @ApiStatus.Internal
+    public SpawnPlacementRegisterEvent(Map<EntityType<?>, MergedSpawnPredicate> map)
+    {
+        this.map = map;
+    }
+
+    /**
+     * Register an optional spawn placement {@code predicate} for a given {@code entityType}
+     */
+    public void register(EntityType<?> entityType, SpawnPlacements.SpawnPredicate<Entity> predicate)
+    {
+        register(entityType, null, null, predicate, Operation.OR);
+    }
+
+    /**
+     * Register a {@code predicate} for a given {@code entityType} with a given {@code operation} for handling
+     */
+    public void register(EntityType<?> entityType, SpawnPlacements.SpawnPredicate<Entity> predicate, Operation operation)
+    {
+        register(entityType, null, null, predicate, operation);
+    }
+
+    /**
+     * Register a {@code predicate} for a given {@code entityType} and {@code operation}
+     * With the option of changing the {@code placementType} and {@code heightmap}.
+     * Use {@code null} for the placement or heightmap to leave them as is (which should be done in almost every case)
+     */
+    public void register(EntityType<?> entityType, @Nullable SpawnPlacements.Type placementType, @Nullable Heightmap.Types heightmap, SpawnPlacements.SpawnPredicate<Entity> predicate, Operation operation)
+    {
+        if (!map.containsKey(entityType))
+        {
+            if (placementType == null)
+            {
+                throw new NullPointerException("Registering a new Spawn Predicate requires a nonnull placement type! Entity Type: " + entityType.getClass().getSimpleName());
+            }
+            if (heightmap == null)
+            {
+                throw new NullPointerException("Registering a new Spawn Predicate requires a nonnull heightmap type! Entity Type: " + entityType.getClass().getSimpleName());
+            }
+            map.put(entityType, new MergedSpawnPredicate(predicate, placementType, heightmap));
+        }
+        else
+        {
+            map.get(entityType).merge(operation, predicate, placementType, heightmap);
+        }
+    }
+
+    @ApiStatus.Internal
+    public void registerAll()
+    {
+        SpawnPlacements.registerAll(map);
+    }
```

Comment:
You don't need this method since the map itself is mutable, so you already have an instance.